shell/platform/android/android_shell_holder.cc

```diff
@@ -135,6 +135,14 @@ AndroidShellHolder::AndroidShellHolder(
 }
 
 AndroidShellHolder::~AndroidShellHolder() {
+  if (platform_view_) {
```

Comment:
The platform view may not have a surface. This may happen in cases of shell termination before the platform view gets a surface. That is not to say that you can just perform a null surface check here however. As a surface could have been obtained after the task was posted. For this reason, I don't this this the right spot to clear the resource context.

Reply:
I think just like `CreateResourceContext` on `PlatformView`, we can have a `DestroyResourceContext`. Currently, the resource context is create when the IO manager is setup (on the IO thread). We know that the IO manager will be collected on the IO thread, so lets collect the resource context at that moment as well. We wont have to deal with timing issues as those have already been sorted out for the IO manager. Also, we will get the exact same behavior on all platforms (though not all need it apparently).

Reply:
Moved this task up to the PlatformView.  PTAL